bundles/org.openhab.binding.irobot/src/main/java/org/openhab/binding/irobot/internal/utils/LoginRequester.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.irobot.internal.utils;
+
+import static org.openhab.binding.irobot.internal.IRobotBindingConstants.MQTT_PORT;
+import static org.openhab.binding.irobot.internal.IRobotBindingConstants.TRUST_MANAGERS;
+import static org.openhab.binding.irobot.internal.IRobotBindingConstants.UDP_PORT;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.StringReader;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+
+import javax.net.ssl.SSLContext;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.irobot.internal.discovery.IRobotDiscoveryService;
+import org.openhab.binding.irobot.internal.dto.MQTTProtocol.BlidResponse;
+
+import com.google.gson.Gson;
+import com.google.gson.stream.JsonReader;
+
+/**
+ * Helper functions to get blid and password. Seems pretty much reinventing a bicycle,
+ * but it looks like HiveMq doesn't provide for sending and receiving custom packets.
+ * The {@link LoginRequester#getBlid} and {@link IRobotDiscoveryService} are heavily
+ * related to each other.
+ *
+ * @author Pavel Fedin - Initial contribution
+ * @author Alexander Falkenstern - Fix password fetching
+ *
+ */
+@NonNullByDefault
+public class LoginRequester {
+    public static @Nullable String getBlid(final String ip) throws IOException {
+        DatagramSocket socket = new DatagramSocket();
+        socket.setSoTimeout(1000); // One second
+        socket.setReuseAddress(true);
+
+        final byte[] bRequest = "irobotmcs".getBytes(StandardCharsets.UTF_8);
+        DatagramPacket request = new DatagramPacket(bRequest, bRequest.length, InetAddress.getByName(ip), UDP_PORT);
+        socket.send(request);
+
+        byte[] reply = new byte[1024];
+        try {
+            DatagramPacket packet = new DatagramPacket(reply, reply.length);
+            socket.receive(packet);
+            reply = Arrays.copyOfRange(packet.getData(), packet.getOffset(), packet.getLength());
+        } catch (IOException exception) {
+        } finally {
```

Comment:
You could add a comment, why this exception is ignored.

Reply:
removed catch-block, as the exception is handled outside the function anyway